File: examples/docs-code-snippets/src/main/scala/examples/scala/application/RegisterBakerEventListener.scala
package examples.scala.application

import com.ing.baker.runtime.scaladsl.Baker

class RegisterBakerEventListener(val baker: Baker) {

  def example(): Unit = {
    baker.registerBakerEventListener((bakerEvent) =>
      println(s"Received event: $bakerEvent")
    )
  }
}

File: examples/docs-code-snippets/src/main/java/examples/java/application/RegisterBakerEventListener.java
package examples.java.application;

import com.ing.baker.runtime.javadsl.Baker;
import com.ing.baker.runtime.javadsl.BakerEvent;

public class RegisterBakerEventListener {

    private final Baker baker;

    public RegisterBakerEventListener(Baker baker) {
        this.baker = baker;
    }

    public void example() {
        baker.registerBakerEventListener((BakerEvent event) ->
            System.out.println("Received event: " + event)
        );
    }
}
